py/visdom/__init__.py

```diff
@@ -248,25 +249,30 @@ def _assert_opts(opts):
     if opts.get('title'):
         assert isstr(opts.get('title')), 'title should be a string'
 
+torch_types = []
+try:
+    import torch
+    torch_types.append(torch.Tensor)
+    torch_types.append(torch.nn.Parameter)
+    torch_types.append(torch.autograd.Variable)
+except (ImportError, AttributeError):
+    pass
 
-def pytorch_wrap(fn):
-    def result(*args, **kwargs):
-        args = (a.cpu().detach().numpy() if type(a).__module__.startswith('torch') else a for a in args)
-
-        for k in kwargs:
-            if type(kwargs[k]).__module__.startswith('torch'):
-                kwargs[k] = kwargs[k].cpu().detach().numpy()
-
-        return fn(*args, **kwargs)
-    return result
-
+def _torch_to_numpy(a):
+    for kind in torch_types:
+        if isinstance(a, kind):
+            return a.cpu().detach().numpy()
```

Comment:
This will fail for PyTorch <0.4 `Variable`s: for those, the tensor is wrapped in `a.data`. To remain backwards compatible, perhaps you can check `hasattr(a, 'data')` to see whether you need to get that field? (Please add a comment stating which bits are for backwards compatibility and a deprecated warning -- that makes it easy for us to remove the <0.4 code later.)

Reply:
The `hasattr` method won't work since 'data' is defined on non-variables in PyTorch 0.3; it will just raise an exception if you try to access it. I tried in my subsequent commit to come up with the most readable version of backwards compatibility and successfully tested it in a virtualenv with PyTorch 0.3.1.

Feel free to push to this PR or edit after merging  if you want to tweak the wording of anything.